web/src/main/java/org/fao/geonet/monitor/link/LinkMonitorService.java

```diff
@@ -0,0 +1,142 @@
+package org.fao.geonet.monitor.link;
+
+import jeeves.interfaces.Logger;
+import jeeves.server.resources.ResourceManager;
+import jeeves.utils.Log;
+import org.apache.lucene.document.DocumentStoredFieldVisitor;
+import org.fao.geonet.GeonetContext;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.kernel.search.IndexAndTaxonomy;
+import org.fao.geonet.kernel.search.index.GeonetworkMultiReader;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LinkMonitorService extends LinkMonitorInterface {
+
+    public enum Status {
+        FAILED,
+        WORKING,
+        UNKNOWN
+    }
+
+    public static int percentWorking = 90;
+
+    public static int maxChecks = 10;
+
+    public static int timeout = 15;
+
+    public static int freshness = 60 * 60;
+
+    private boolean unknownAsWorking = true;
+
+    // Milliseconds between running checks on every record. This is here to
+    // prevent undesired hammering of servers
+    private static int betweenChecksIntervalMs = 100;
+
+    private final Map<String, MetadataRecordInfo> recordMap;
+
+    private long reindexTimestamp = -1;
+
+    public LinkMonitorService() {
+        this.recordMap = new HashMap<String, MetadataRecordInfo>();
+    }
+
+    @Override
+    public void init(ResourceManager resourceManager, GeonetContext geonetContext, Logger logger, long reindexInterval) {
+        super.init(resourceManager, geonetContext, logger, reindexInterval);
+    }
+
+    private ArrayList<String> getAllRecords() {
+        try {
+            return getAllRecordsLucene();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
The result of this method is used (in `reindex()`) without a null check.  Therefore, a NPE lies in wait.

Under what circumstances can `getAllRecordsLucene()` throw an exception?  Maybe better for this method to return an empty list in this case?